### ISSUE TITLE
Failed test with zend-form 2.7

### ISSUE THREAD

**Issue by remicollet** [zendframework/zend-mvc#72]
With zend-form 2.6, test suite is ok

With zend-form 2.7

There were 2 errors:

```
1) ZendTest\Mvc\Service\FormAnnotationBuilderFactoryTest::testCreateService
Missing argument 2 for Mock_FormElementManager_1b518e1e::injectFactory(), called in /builddir/build/BUILDROOT/php-zendframework-zend-mvc-2.6.2-1.fc20.remi.x86_64/usr/share/php/Zend/Mvc/Service/FormAnnotationBuilderFactory.php on line 32 and defined

/builddir/build/BUILDROOT/php-zendframework-zend-mvc-2.6.2-1.fc20.remi.x86_64/usr/share/php/Zend/Mvc/Service/FormAnnotationBuilderFactory.php:32
/builddir/build/BUILD/zend-mvc-9babf3f501f7db532f3293ec39593237f297a7d0/test/Service/FormAnnotationBuilderFactoryTest.php:28

2) ZendTest\Mvc\Service\FormAnnotationBuilderFactoryTest::testCreateServiceSetsPreserveDefinedOrder
Missing argument 2 for Mock_FormElementManager_1b518e1e::injectFactory(), called in /builddir/build/BUILDROOT/php-zendframework-zend-mvc-2.6.2-1.fc20.remi.x86_64/usr/share/php/Zend/Mvc/Service/FormAnnotationBuilderFactory.php on line 32 and defined

/builddir/build/BUILDROOT/php-zendframework-zend-mvc-2.6.2-1.fc20.remi.x86_64/usr/share/php/Zend/Mvc/Service/FormAnnotationBuilderFactory.php:32
/builddir/build/BUILD/zend-mvc-9babf3f501f7db532f3293ec39593237f297a7d0/test/Service/FormAnnotationBuilderFactoryTest.php:42
```

**Comment by weierophinney** [comment]
Confirmed. The issue is that initializers are expected to receive two arguments, but that factory is only using one; the tests are incorrect. Essentially, the zend-form update exposed a problem in the zend-mvc test suite!

**Comment by weierophinney** [comment]
Or, rather, the `FormAnnotationBuilderFactory` is incorrect in this case, as it was calling an initializer method without all required arguments. The zend-form v2.6 and previous versions of the initializer omitted the second argument out of convenience, but should not have. Since that's been corrected, the usage in the zend-mvc factory needs to be updated as well. Will push a fix in a minute.

**Comment by remicollet** [comment]
Please just fix the release description for state 2.6.4 when tag state 2.6.3 ;)

**Comment by weierophinney** [comment]
@remicollet It was released with 2.6.3. :)

**Comment by remicollet** [comment]
But https://github.com/zendframework/zend-mvc/releases/tag/release-2.6.3 says **zend-mvc 2.6.4** ;)
